pyquil/external/rpcq.py

```diff
@@ -1,56 +1,163 @@
+import json
 from typing import Dict, List, Union, Optional, Any
-from typing_extensions import Literal
-from pydantic import BaseModel, Field
 from rpcq.messages import TargetDevice as TargetQuantumProcessor
+from dataclasses import dataclass, field
 
+#JsonValue = Union[type(None), bool, int, float, str, list["JsonValue"], dict[str, "JsonValue"]]
+JsonValue = Any
 
-class Operator(BaseModel):
+@dataclass
+class Operator:
     operator: Optional[str] = None
     duration: Optional[float] = None
     fidelity: Optional[float] = None
 
-
+    def __post_init__(self):
+        self.duration = float(self.duration) if self.duration is not None else None
+        self.fidelity = float(self.fidelity) if self.fidelity is not None else None
+
+    def to_dict(self) -> Dict[str, JsonValue]:
```

Comment:
`asdict` _might_ be workable for some of these classes, but there was a bunch of customization of the json logic in a few of them that seems to me to require a hand-written method.

Reply:
Ah, yeah, the Qubit customization is tricky. I think it is fine to keep, but for compatibility with the former pydantic API, lets rename `to_dict` to `dict` and `from_dict` to `parse_obj` and deprecate each, since I also don't think they are used anywhere.

Reply:
I believe I have now done this the way you describe.  I gave each class private `_dict` and`_parse_obj` methods so that the recursive calls descending into the data structure don't produce hundreds of deprecation warnings when you serialize the top-level object.  The public `dict` and `parse_obj` are deprecated and simply delegate to the private ones.